### ISSUE TITLE
argMax → StableHLO lowering emits invalid IR (int-literal f32 constant + non-collapsing reduce)

### ISSUE THREAD

**Issue by michalharakal** [sk-ai-net/skainet#876]
The `argMax` dag{} builder (core 0.35.0+) exports, but its StableHLO lowering produces IR that
`iree-compile` (stock IREE 3.11.0, `--iree-input-type=stablehlo`, llvm-cpu) rejects. Found via the
`functiongemma-270m` conformance row (skainet-iree-conformance #24) at the 0.36.0 pin.

**Repro:** `argMax(logits, axis=3, ...)` over a `tensor<1x1x4x256xf32>`. Emitted tail:

```mlir
%v39 = stablehlo.reduce(%v37 init: %v38) applies stablehlo.maximum across dimensions = [3]
       : (tensor<1x1x4x256xf32>, tensor<f32>) -> tensor<1x1x4xf32>
%v40 = stablehlo.broadcast_in_dim %v39, dims = [0, 1, 2] : (tensor<1x1x4xf32>) -> tensor<1x1x4x256xf32>
%v41 = stablehlo.compare EQ, %v37, %v40 : (...) -> tensor<1x1x4x256xi1>
%v42 = stablehlo.iota dim = 3 : tensor<1x1x4x256xf32>
%v43 = stablehlo.constant dense<256> : tensor<1x1x4x256xf32>          # BUG 1
%v44 = stablehlo.select %v41, %v42, %v43 : tensor<1x1x4x256xi1>, tensor<1x1x4x256xf32>
%v45 = stablehlo.constant dense<256> : tensor<f32>
%v46 = stablehlo.reduce(%v44 init: %v45) applies stablehlo.minimum across dimensions = [3]
       : (tensor<1x1x4x256xf32>, tensor<f32>) -> tensor<1x1x4x256xf32>   # BUG 2
return %v46 : tensor<1x1x4x256xf32>
```

**Bug 1 (blocks compile):** an f32 constant is emitted as an **integer literal** —
`stablehlo.constant dense<256> : tensor<1x1x4x256xf32>`. iree-compile:
```
error: unexpected decimal integer literal for a floating point value
    %v43 = stablehlo.constant dense<256> : tensor<1x1x4x256xf32>
note: add a trailing dot to make the literal a float
```
Should be `dense<2.560000e+02>` (or `256.0`).

**Bug 2 (shape):** the final `stablehlo.reduce` across dim `[3]` declares output
`tensor<1x1x4x256xf32>` — it does **not collapse** the reduced dim. A reduce over dim 3 of a
`1x1x4x256` input must yield `1x1x4`. As-is the "argmax" result keeps the vocab dim and is f32, not the
expected token-id shape/dtype.

**Expected:** argMax lowers to compilable StableHLO whose result drops the reduced axis (token indices).

**Impact:** blocks the FunctionGemma functional-token decode tail (argMax over lm_head logits) from
compiling — the shipping SL2610 demo's decode. Conformance row: `functiongemma-270m` (Define/Export ok,
Compile red) in skainet-iree-conformance.
